validation/src/header.rs

```diff
@@ -607,4 +620,97 @@ mod test {
             Err(ValidateHeaderError::TargetDifficultyAboveMax)
         ));
     }
+
+    fn genesis_header(bits: u32) -> BlockHeader {
+        BlockHeader {
+            version: 1,
+            prev_blockhash: Default::default(),
+            merkle_root: Default::default(),
+            time: 1296688602,
+            bits,
+            nonce: 0,
+        }
+    }
+
+    fn next_block_header(prev: BlockHeader, bits: u32) -> BlockHeader {
+        BlockHeader {
+            prev_blockhash: prev.block_hash(),
+            time: prev.time + TEN_MINUTES,
+            bits,
+            ..prev
+        }
+    }
+
+    /// Creates a chain of headers with the given length and
+    /// proof of work for the first header.
+    fn create_chain(
+        network: &Network,
+        initial_pow: u32,
+        chain_length: u32,
+    ) -> (SimpleHeaderStore, BlockHeader) {
+        let pow_limit = pow_limit_bits(network);
+        let h0 = genesis_header(initial_pow);
+        let mut store = SimpleHeaderStore::new(h0, 0);
+        let mut last_header = h0;
+
+        for _ in 1..chain_length {
+            let new_header = next_block_header(last_header, pow_limit);
+            store.add(new_header);
+            last_header = new_header;
+        }
+
+        (store, last_header)
+    }
+
+    const SOME_NON_LIMIT_POW: u32 = 7;
+    const CHAIN_LENGTH_MAX: u32 = 10;
+
+    #[test]
+    fn test_find_next_difficulty_in_chain_pow_found() {
```

Comment:
I'm not sure these tests are adding coverage, since they pass without the bug fixes of this PR. I think we should have at least one test case that fails before your change, and now passes with your change.

Reply:
I agree that `test_find_next_difficulty_in_chain_pow_not_found` do not fail without those changes. But the ones you pointed out `test_find_next_difficulty_in_chain_pow_found` do fail without those changes.

I removed `test_find_next_difficulty_in_chain_pow_not_found`.